File: Modules/ServerTypes/TaskManager/TaskColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TaskPriority } from "./TaskPriority";
import { TaskRow } from "./TaskRow";

export interface TaskColumns {
    TaskId: Column<TaskRow>;
    Title: Column<TaskRow>;
    Description: Column<TaskRow>;
    DueDate: Column<TaskRow>;
    Priority: Column<TaskRow>;
    IsCompleted: Column<TaskRow>;
    InsertDate: Column<TaskRow>;
}

export class TaskColumns extends ColumnsBase<TaskRow> {
    static readonly columnsKey = 'TaskManager.Task';
    static readonly Fields = fieldsProxy<TaskColumns>();
}

[TaskPriority]; // referenced types

File: Modules/TaskManager/Task/TaskColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace TaskManager.TaskManager.Columns;

[ColumnsScript("TaskManager.Task")]
[BasedOnRow(typeof(TaskRow), CheckNames = true)]
public class TaskColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int TaskId { get; set; }
    [EditLink]
    public string Title { get; set; }
    public string Description { get; set; }
    public DateTime DueDate { get; set; }
    public int Priority { get; set; }
    public bool IsCompleted { get; set; }
    public DateTime InsertDate { get; set; }
}